src/lib/internal/Lexer.spec.ts

```diff
@@ -76,7 +76,9 @@ describe("class Lexer", () => {
           const expected = [
             ...row.flatMap((field, i) => [
               // if field is empty, it should be ignored
-              ...(field !== "" ? [{ type: Field, value: field }] : []),
+              ...(field !== "" || escapeField(field, { delimiter }) !== field
+                ? [{ type: Field, value: field }]
+                : []),
```

Comment:
Since escapeField does not escape by default and returns escaped results when necessary, the difference between the result of escapeField and the value of field is the answer to whether the field needs to be escaped or not.